core/tests/protractor_utils/users.js

```diff
@@ -87,7 +87,8 @@ var _completeSignup = async function(username) {
   await waitFor.pageToFullyLoad();
   let cookieBannerAcceptButton = element(
     by.css('.protractor-test-oppia-cookie-banner-accept-button'));
-  if (await cookieBannerAcceptButton.isPresent()) {
+  let cookie = await browser.manage().getCookie('OPPIA_COOKIES_ACKNOWLEDGED');
+  if (!cookie) {
```

Comment:
This is to make sure that we accept the banner if it shows up. Previously, `isPresent()` would return false if the banner shows up later on -- this caused the cookie banner to show throughout the test.

Reply:
I have two doubts.

1. Suppose we get the `cookie` from the console using `browser.manage().getCookie ` but as pointed by you above, it might happen that the cookie banner appears after sometime. So, then clicking on the cookie banner accept button might create an error.

2. Why are we checking `!cookie` in the if condition? (Not Sure thought. Can  you please explain this a bit?)

Reply:
@SAEb-ai, the way the cookie banner works is, if there is no cookie, the banner shows up and it is hidden if the cookie exists. The statement `await browser.manage().getCookie('OPPIA_COOKIES_ACKNOWLEDGED')` returns an object if the cookie exists and `null` otherwise. If it exists, `!cookie` will be false and not wait for the cookie banner. If the cookie is not found, `!cookie` will be true and we wait for the cookie banner and click on the button to dismiss it. Hope this clarifies! Thanks!

Reply:
Thanks! It clearedy doubts.